compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/expr.rs

```diff
@@ -416,6 +416,32 @@ impl Expr {
         }
     }
 
+    // Attempts automatic casting of arguments to type of function parameters
+    pub fn cast_function_arguments(
```

Comment:
This should have an explanation of why we want to do this, and maybe a name that better explains what we're doing (converting between abstract and concrete types iff they're the same on the given machine)

Reply:
Fixed